R/function_options.R

```diff
@@ -0,0 +1,69 @@
+extendr_function_config <- rlang::env(
+  known_options = tibble::tribble(
+    ~Name, ~Ptype,
+    "use_try_from", logical(),
+    "r_name", character(),
+    "mod_name", character(),
+  )
+)
+
+convert_function_options <- function(options, quiet) {
+  if (rlang::is_null(options) || rlang::is_empty(options)) {
+    return(tibble::tibble(Name = character(), RustValue = character()))
+  }
+
+  if (!rlang::is_list(options) || !rlang::is_named(options)) {
+    cli::cli_abort("Extendr function options should be either a named {.code list()} or {.code NULL}.")
+  }
+
+  options_table <- tibble::tibble(Name = rlang::names2(options), Value = unname(options)) %>%
+    dplyr::left_join(extendr_function_config$known_options, by = "Name") %>%
+    dplyr::mutate(
+      Value = purrr::pmap(
+        list(.data$Value, .data$Ptype, .data$Name),
+        ~ if (rlang::is_null(..2)) ..1 else vctrs::vec_cast(..1, ..2, x_arg = ..3)
+      ),
+    )
```

Comment:
Is there a way to simplify this using [`dplyr::rows_update`](https://dplyr.tidyverse.org/reference/rows.html)?